File: examples/take-a-screenshot/rest/java/TakeScreenshot.java
// Captures a webpage as a PNG using the Browserless /screenshot endpoint.
// java.net.http is included in JDK 11+ — no extra dependency needed.
//
// Run: javac TakeScreenshot.java && java TakeScreenshot

import java.io.*;
import java.net.URI;
import java.net.http.*;

public class TakeScreenshot {
    private static final String TOKEN = "YOUR_API_TOKEN_HERE";

    public static void main(String[] args) throws Exception {
        String endpoint = "https://production-sfo.browserless.io/screenshot?token=" + TOKEN;
        String jsonData = """
            {
                "url": "https://example.com",
                "options": { "fullPage": true, "type": "png" }
            }""";

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint))
            .header("Cache-Control", "no-cache")
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(jsonData))
            .build();

        HttpResponse<InputStream> response = client.send(request, HttpResponse.BodyHandlers.ofInputStream());
        try (FileOutputStream fos = new FileOutputStream("screenshot.png")) {
            response.body().transferTo(fos);
        }
        System.out.println("Screenshot saved as screenshot.png.");
    }
}

File: examples/take-a-screenshot/rest/csharp/TakeScreenshot.cs
// Captures a webpage as a PNG using the Browserless /screenshot endpoint.
// Uses System.Net.Http and System.Text.Json from the .NET standard library.
//
// Run: dotnet run

using System.Net.Http;
using System.Text;
using System.Text.Json;

const string Token = "YOUR_API_TOKEN_HERE";
string endpoint = $"https://production-sfo.browserless.io/screenshot?token={Token}";

var payload = new
{
    url = "https://scraping-sandbox.netlify.app/receipt",
    options = new { fullPage = true, type = "png" },
};

using var client = new HttpClient();
var request = new HttpRequestMessage(HttpMethod.Post, endpoint);
request.Headers.Add("Cache-Control", "no-cache");
request.Content = new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json");

var response = await client.SendAsync(request);
var contentStream = await response.Content.ReadAsStreamAsync();
using var fileStream = new FileStream("screenshot.png", FileMode.Create, FileAccess.Write);
await contentStream.CopyToAsync(fileStream);
Console.WriteLine("Screenshot saved as screenshot.png.");
